### PR TITLE
feat:(prl-3887 : added intial setup for CosApiClient,GetController on…

### PR DESCRIPTION
**Pull request by SasankAchyuta9**
… hearings)

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] the keep-helm label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/app/case/CosApiClient.ts [hunk 1]
```diff
@@ -139,13 +143,45 @@ export class CosApiClient {
...
+      console.log(`Auth Token ****** ${serviceauth} *******`);
 
       return { id: response.data.id, state: response.data.state, ...fromApiFormat(response.data) };
     } catch (err) {
       throw new Error('Case could not be updated.');
     }
   }
 
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  public async getAllHearingsForCitizenCase(user: UserDetails, caseId: string): Promise<any> {
```
Comment (vivek-sekhar) [inline src/main/app/case/CosApiClient.ts:159]: New method is not required. Change the existing method to pull hearing details from COS

File: src/main/app/case/definition.ts [hunk 2]
```diff
@@ -2395,12 +2398,65 @@ export interface PRLDocument {
...
+}
+
+export interface futureHearing{
+    Dates?: string,
+    hearingLength?: string,
+    hearingMethod?: string,
+    schedules?: futSchedule[]
+}
+
+export interface futSchedule{
```
Comment (vivek-sekhar) [inline src/main/app/case/definition.ts:2444]: Why is this required?

File: src/main/app/case/definition.ts [hunk 3]
```diff
@@ -2395,12 +2398,65 @@ export interface PRLDocument {
...
+  hearingVenueAddress?: string | null,
+  hearingRoomId?: string | null,
+  hearingJudgeId?: string | null,
+  hearingJudgeName?: string | null,
+  panelMemberIds?: string[] | Number[] | null,
+  attendees?: Attendee[] | null,
+
+}
+
+export interface futureHearing{
```
Comment (vivek-sekhar) [inline src/main/app/case/definition.ts:2437]: Why is this required?

File: src/main/steps/applicant/applicantCaseSequence.ts [hunk 4]
```diff
@@ -551,4 +551,9 @@ export const applicantCaseSequence: Step[] = [
     showInSection: Sections.AboutApplicantCase,
     getNextStep: () => APPLICANT_TASKLIST_HEARING_NEEDS,
   },
+  {
```
Comment (vivek-sekhar) [inline src/main/steps/applicant/applicantCaseSequence.ts:554]: I don't think it is required.

File: src/main/steps/applicant/task-list/get.ts [hunk 5]
```diff
@@ -1,9 +1,250 @@
...
 export default class ApplicantTaskListGetController extends GetController {
   constructor() {
     super(__dirname + '/template', generateContent);
   }
+  public async load(req: AppRequest, res: Response): Promise<void> {
+    // const citizenUser = req.session.user;
+    // const caseId = req.session.userCase.id;
+    // const client = new CosApiClient(citizenUser.accessToken, 'https://return-url');
+    // const hearings = await client.getAllHearingsForCitizenCase(citizenUser, caseId);
+    req.session.userCase.hearingCollection = [
```
Comment (vivek-sekhar) [inline src/main/steps/applicant/task-list/get.ts:22]: Please refer PRl-3914 PR for code reference to load hearing data

File: src/main/steps/applicant/task-list/tasklist.ts [hunk 6]
```diff
@@ -54,8 +56,11 @@ export const generateApplicantTaskList = (sectionTitles, taskListItems, userCase
         {
           id: 'check-details-of-your-court-hearings',
           text: taskListItems.details_of_court_hearings,
-          status: getApplicantViewAllHearingsFromTheCourt(userCase),
-          href: URL.APPLICANT_YOURHEARINGS_HEARINGS,
+          status: getViewAllHearingsFromTheCourt(userCase),
+          href:
+            getViewAllHearingsFromTheCourt(userCase) === 'READY_TO_VIEW'
```
Comment (vivek-sekhar) [inline src/main/steps/applicant/task-list/tasklist.ts:61]: Disable the link if case hearings is not applicable.

File: src/main/steps/applicant/task-list/tasklist.ts [hunk 7]
```diff
@@ -85,7 +90,10 @@ export const generateApplicantTaskList = (sectionTitles, taskListItems, userCase
           id: 'view-all-orders-from-the-court',
           text: taskListItems.view_all_orders_from_the_court,
           status: getViewAllOrdersFromTheCourt(userCase),
-          href: getViewAllOrdersFromTheCourt(userCase) === 'READY_TO_VIEW' ? URL.APPLICANT_ORDERS_FROM_THE_COURT : '#',
+          href:
+            getViewAllOrdersFromTheCourt(userCase) === 'READY_TO_VIEW'
+              ? URL.APPLICANT_ORDERS_FROM_THE_COURT + '/' + userCase.id
```
Comment (vivek-sekhar) [inline src/main/steps/applicant/task-list/tasklist.ts:95]: don't use string concatenation . use string literals.

File: src/main/steps/applicant/yourhearings/hearings/content.ts [hunk 8]
```diff
@@ -1,10 +1,30 @@
...
+import { APPLICANT_TASK_LIST_URL, FETCH_CASE_DETAILS } from '../../../../steps/urls';
+import { CaseType, PartyType } from '../../../../app/case/definition';
 
 export { form } from '../../../common/yourhearings/hearings/content';
 export const generateContent: TranslationFn = content => {
   const hearingsContent = yourhearingshearingscontent(content);
+  const request = content.additionalData?.req;
+  const caseData = request.session.userCase;
+  if (content.additionalData?.req.session.userCase.caseTypeOfApplication === 'C100') {
+    hearingsContent.linkforsupport = '/applicant/hearing-needs/support-help';
```
Comment (vivek-sekhar) [inline src/main/steps/applicant/yourhearings/hearings/content.ts:13]: Don't hard code ur's here. Use enums for condition checking

File: src/main/steps/applicant/yourhearings/hearings/content.ts [hunk 9]
```diff
@@ -1,10 +1,30 @@
...
 import { generateContent as yourhearingshearingscontent } from '../../../common/yourhearings/hearings/content';
+import { APPLICANT_TASK_LIST_URL, FETCH_CASE_DETAILS } from '../../../../steps/urls';
+import { CaseType, PartyType } from '../../../../app/case/definition';
 
 export { form } from '../../../common/yourhearings/hearings/content';
 export const generateContent: TranslationFn = content => {
   const hearingsContent = yourhearingshearingscontent(content);
+  const request = content.additionalData?.req;
+  const caseData = request.session.userCase;
+  if (content.additionalData?.req.session.userCase.caseTypeOfApplication === 'C100') {
```
Comment (vivek-sekhar) [inline src/main/steps/applicant/yourhearings/hearings/content.ts:12]: Content.ts should have the logic to build the case data according to design.

File: src/main/steps/common/task-list/controllers/CaseDetailsGetController.ts [hunk 10]
```diff
@@ -16,11 +16,205 @@ export default class CaseDetailsGetController {
     }
 
     try {
+      const data = [
```
Comment (vivek-sekhar) [inline src/main/steps/common/task-list/controllers/CaseDetailsGetController.ts:19]: Please refer PRl-3914 PR for code reference to load hearing data

File: src/main/steps/common/yourhearings/hearings/HearingsGetController.ts [hunk 11]
```diff
@@ -1,55 +1,320 @@
...
+    }
+
+    req.session.save(() => res.redirect(redirectUrl));
+    console.log('I am in common hearings get controller');
+  }
+}
+
+export function getMonthName(month: number, language:string): string {
+  let monthToDisplay = '';
+  if(language === 'cy'){
```
Comment (vivek-sekhar) [inline src/main/steps/common/yourhearings/hearings/HearingsGetController.ts:157]: use month translations from translation infra

File: src/main/steps/common/yourhearings/hearings/HearingsGetController.ts [hunk 12]
```diff
@@ -1,55 +1,320 @@
...
-    res.render(this.view, {
-      ...content,
-      sessionErrors,
-      htmlLang: language,
-      formaction,
-      userIdamId: req.session?.user?.id,
-      ordinalNumberMap,
-      userCase,
-    });
+    const partyType = getCasePartyType(req.session.userCase, req.session.user.id);
```
Comment (vivek-sekhar) [inline src/main/steps/common/yourhearings/hearings/HearingsGetController.ts:16]: Shift all template data logic to content.ts and just redirect to the respective page.

File: src/main/steps/respondent/respondentcaseSequence.ts [hunk 13]
```diff
@@ -520,4 +518,9 @@ export const respondentCaseSequence: Step[] = [
     showInSection: Sections.AboutRespondentCase,
     getNextStep: () => RESPONDENT_VIEW_ALL_DOCUMENTS,
   },
+  {
+    url: RESPONDENT_TASK_LIST_URL,
```
Comment (vivek-sekhar) [inline src/main/steps/respondent/respondentcaseSequence.ts:522]: Why is this required?

File: src/main/steps/respondent/yourhearings/hearings/content.ts [hunk 14]
```diff
@@ -1,10 +1,19 @@
...
 
 export { form } from '../../../common/yourhearings/hearings/content';
 export const generateContent: TranslationFn = content => {
   const hearingsContent = yourhearingshearingscontent(content);
+  hearingsContent.linkforsupport = '/respondent/support-you-need-during-case/attending-the-court';
+  const request = content.additionalData?.req;
+  const caseData = request.session.userCase;
   return {
     ...hearingsContent,
+    breadcrumb:{
```
Comment (vivek-sekhar) [inline src/main/steps/respondent/yourhearings/hearings/content.ts:14]: Breadcrumb shd be shown only for CA applicant journey and for all other journey it should be just back navigation.

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/app/case/CosApiClient.ts (hunk 1 above)
Comment (SasankAchyuta9) [reply src/main/app/case/CosApiClient.ts:159]: Yeah using the old mehtod now.

File: src/main/app/case/definition.ts (hunk 2 above)
Comment (SasankAchyuta9) [reply src/main/app/case/definition.ts:2444]: Yes removed this.

File: src/main/app/case/definition.ts (hunk 3 above)
Comment (SasankAchyuta9) [reply src/main/app/case/definition.ts:2437]: yes removed.

File: src/main/steps/respondent/yourhearings/hearings/content.ts (hunk 14 above)
Comment (SasankAchyuta9) [reply src/main/steps/respondent/yourhearings/hearings/content.ts:14]: Yes this has been taken care now.

File: src/main/steps/applicant/task-list/tasklist.ts (hunk 7 above)
Comment (SasankAchyuta9) [reply src/main/steps/applicant/task-list/tasklist.ts:95]: done this.

File: src/main/steps/applicant/task-list/get.ts (hunk 5 above)
Comment (SasankAchyuta9) [reply src/main/steps/applicant/task-list/get.ts:22]: done this.

File: src/main/steps/common/task-list/controllers/CaseDetailsGetController.ts (hunk 10 above)
Comment (SasankAchyuta9) [reply src/main/steps/common/task-list/controllers/CaseDetailsGetController.ts:19]: done this.

File: src/main/steps/respondent/respondentcaseSequence.ts (hunk 13 above)
Comment (SasankAchyuta9) [reply src/main/steps/respondent/respondentcaseSequence.ts:522]: yes removed this.

File: src/main/steps/applicant/yourhearings/hearings/content.ts (hunk 9 above)
Comment (SasankAchyuta9) [reply src/main/steps/applicant/yourhearings/hearings/content.ts:12]: Yeah this for link on where support during case should be navigated to . And it is given on content.ts itself.

File: src/main/steps/applicant/yourhearings/hearings/content.ts (hunk 8 above)
Comment (SasankAchyuta9) [reply src/main/steps/applicant/yourhearings/hearings/content.ts:13]: Yeah corrected this.

File: src/main/steps/common/yourhearings/hearings/HearingsGetController.ts (hunk 12 above)
Comment (SasankAchyuta9) [reply src/main/steps/common/yourhearings/hearings/HearingsGetController.ts:15]: done this.

File: src/main/app/case/CosApiClient.ts [hunk 15]
```diff
@@ -419,49 +392,32 @@ export class CosApiClient {
...
-  //     return {
-  // id: userCase.id,
-  // state: userCase.state,
-  // hearingCollection: hearingCollectionMockedData,
-  //     };
-  //   }
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  public async retrieveCaseHearingsByCaseId(user: UserDetails, caseId: string): Promise<any> {
+    const path = config.get('services.cos.url') + `/hearing/${caseId}`;
+    logger.info(`Path ****** ${path} *******`);
```
Comment (vivek-sekhar) [inline src/main/app/case/CosApiClient.ts:398]: Remove loggers and consoles

File: src/main/steps/common/yourhearings/hearings/content.ts [hunk 16]
```diff
@@ -135,8 +140,313 @@ export const form: FormContent = {
...
+    }
+    // console.log("8***********" , partyData);
+    if (hearingMethod !== null) {
+      return hearingMethod;
+    }
+    return '';
+  }
+  
+  export function getProperTime(date: Date): string {
+    let dateString;
```
Comment (vivek-sekhar) [inline src/main/steps/common/yourhearings/hearings/content.ts:440]: Use dayJS for all date formatting.

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/applicant/task-list/tasklist.ts [hunk 17]
```diff
@@ -92,7 +94,7 @@ export const generateApplicantTaskList = (sectionTitles, taskListItems, userCase
           status: getViewAllOrdersFromTheCourt(userCase),
           href:
             getViewAllOrdersFromTheCourt(userCase) === 'READY_TO_VIEW'
-              ? URL.APPLICANT_ORDERS_FROM_THE_COURT + '/' + userCase.id
+              ? `${URL.APPLICANT_ORDERS_FROM_THE_COURT}+${userCase.id}`
```
[the remark this answers is not in the document]
Comment (SasankAchyuta9) [reply src/main/steps/applicant/task-list/tasklist.ts:97]: Yes vivek , this is working. and same kind of thing is being used at multiple places like orders from court...etc..

File: src/main/steps/applicant/yourhearings/hearings/content.ts [hunk 18]
```diff
@@ -1,15 +1,15 @@
...
 export { form } from '../../../common/yourhearings/hearings/content';
 export const generateContent: TranslationFn = content => {
   const hearingsContent = yourhearingshearingscontent(content);
   const request = content.additionalData?.req;
   const caseData = request.session.userCase;
-  if (content.additionalData?.req.session.userCase.caseTypeOfApplication === 'C100') {
+  if (content.additionalData?.req.session.userCase.caseTypeOfApplication === CaseType.C100) {
     hearingsContent.linkforsupport = '/applicant/hearing-needs/support-help';
   } else {
     hearingsContent.linkforsupport = '/applicant/support-you-need-during-case/attending-the-court';
```
[the remark this answers is not in the document]
Comment (SasankAchyuta9) [reply src/main/steps/applicant/yourhearings/hearings/content.ts:15]: Done this.

File: src/main/steps/common/yourhearings/hearings/content.ts [hunk 19]
```diff
@@ -135,8 +140,313 @@ export const form: FormContent = {
...
+
+const updateCaseDataForHearings = (req:AppRequest<Partial<Case>>):void =>  {
+    req.session.userCase.nextHearing1 = [];
+    req.session.userCase.futureHearings1 = [];
+    req.session.userCase.completedHearings1 = [];
+    let HearingsFuture:HearingsList[] = [];
+    let HearingsCompleted:HearingsList[] = [];
+    if (req.session.userCase.hearingCollection && req.session.userCase.hearingCollection!.length >= 1) {
+      for (const hearing of req.session.userCase.hearingCollection) {
+          if (hearing.hmcStatus === 'COMPLETED') {
```
[the remark this answers is not in the document]
Comment (SasankAchyuta9) [reply src/main/steps/common/yourhearings/hearings/content.ts:158]: Done this.

File: src/main/app/case/CosApiClient.ts (hunk 15 above)
Comment (SasankAchyuta9) [reply src/main/app/case/CosApiClient.ts:398]: Yes removed.

File: src/main/steps/common/yourhearings/hearings/content.ts (hunk 16 above)
Comment (SasankAchyuta9) [reply src/main/steps/common/yourhearings/hearings/content.ts]: used dayJS

File: src/main/steps/common/yourhearings/hearings/content.ts [hunk 20]
```diff
@@ -135,8 +140,313 @@ export const form: FormContent = {
...
+  }
+  
+  export function getDuration(timeDiff: number): string {
+   if(timeDiff >= 5){
+    return '1day';
+   }
+   return timeDiff.toString();
+  }
+  
+  export function getDayName(day: number , language:string): string {
```
[the remark this answers is not in the document]
Comment (SasankAchyuta9) [reply src/main/steps/common/yourhearings/hearings/content.ts]: used dayJS

File: src/main/steps/common/yourhearings/hearings/content.ts [hunk 21]
```diff
@@ -59,8 +143,164 @@ export const form: FormContent = {
...
+        }
+        req.session.userCase.futureHearings1.push({
+          dates,
+          lengthOfHearing,
+          hearingMethod,
+          hearingDaySchedule: hearingDays,
+        });
+      }
+      const next = req.session.userCase.futureHearings1.shift();
+      req.session.userCase.nextHearing1.push(next!);
```
Comment (vivek-sekhar) [inline src/main/steps/common/yourhearings/hearings/content.ts:235]: why next hearing1

File: src/main/steps/common/yourhearings/hearings/content.ts [hunk 22]
```diff
@@ -59,8 +143,164 @@ export const form: FormContent = {
...
 export const generateContent: TranslationFn = content => {
   const translations = languages[content.language]();
+  updateCaseDataForHearings(content.additionalData?.req);
   return {
     ...translations,
     form: { ...form, fields: (form.fields as FormFieldsFn)(content.userCase || {}) },
   };
 };
+
+const updateCaseDataForHearings = (req: AppRequest<Partial<Case>>): void => {
```
Comment (vivek-sekhar) [inline src/main/steps/common/yourhearings/hearings/content.ts:153]: change the method name to prepareHearingData and return the hearing list as needed in template.njk

Comment (vivek-sekhar) [reply src/main/steps/common/yourhearings/hearings/content.ts:153]: Do not store anything in session

File: src/main/app/controller/GetController.ts [hunk 23]
```diff
@@ -30,6 +30,7 @@ export class GetController {
     const document_type = this.getDocumentType(req);
     const byApplicant = req.query['byApplicant'] as string;
     const addresses = req.session?.addresses;
+
```
Comment (vivek-sekhar) [inline src/main/app/controller/GetController.ts:33]: why extra line?

File: src/main/app/document/DocumentManagementController.ts [hunk 24]
```diff
@@ -64,7 +64,7 @@ export class DocumentManagerController extends PostController<AnyObject> {
       this.fileNameElementMap.set('downloadCitizenDocument', {
         elements: ['citizenUploadedDocumentList', 'citizenDocument'],
       });
-      this.fileNameElementMap.set('orders', { elements: ['orderCollection', 'orderDocument'] });
+      this.fileNameElementMap.set('orders', { elements: ['orderCollection', 'orderDocument', 'orderDocumentWelsh'] });
```
Comment (vivek-sekhar) [inline src/main/app/document/DocumentManagementController.ts:67]: why this is required for your story

File: src/main/steps/applicant/yourhearings/hearings/content.ts [hunk 25]
```diff
@@ -1,10 +1,47 @@
...
+  } else {
+    hearingsContent.linkforsupport = APPLICANT_ATTENDING_THE_COURT;
+  }
+  request.session.userCase.hearingOrders = [];
+  for (const doc of request.session.userCase?.orderCollection || []) {
+    if (doc.value.selectedHearingType) {
+      const uid = doc.value.orderDocument.document_url.substring(
+        doc.value.orderDocument.document_url.lastIndexOf('/') + 1
+      );
+      request.session.userCase.hearingOrders?.push({
```
Comment (vivek-sekhar) [inline src/main/steps/applicant/yourhearings/hearings/content.ts:28]: Don't push anything to usercase if it is required only within template.njk

File: src/main/steps/respondent/task-list/content.ts [hunk 26]
```diff
@@ -1,3 +1,4 @@
+//import { CosApiClient } from '../../../app/case/CosApiClient';
```
Comment (vivek-sekhar) [inline src/main/steps/respondent/task-list/content.ts:1]: why is this line added?

File: src/main/steps/respondent/yourhearings/hearings/content.ts [hunk 27]
```diff
@@ -1,9 +1,27 @@
...
   const hearingsContent = yourhearingshearingscontent(content);
+  hearingsContent.linkforsupport = CA_DA_ATTENDING_THE_COURT;
+  const request = content.additionalData?.req;
+  request.session.userCase.hearingOrders = [];
+  for (const doc of request.session.userCase?.orderCollection || []) {
+    if (doc.value.selectedHearingType) {
+      const uid = doc.value.orderDocument.document_url.substring(
+        doc.value.orderDocument.document_url.lastIndexOf('/') + 1
+      );
+      request.session.userCase.hearingOrders?.push({
```
Comment (vivek-sekhar) [inline src/main/steps/respondent/yourhearings/hearings/content.ts:16]: Why is data pushed to usercase if it is required only in template.njk?

File: src/main/steps/tasklistresponse/TasklistGetController.ts [hunk 28]
```diff
@@ -29,7 +29,7 @@ export class TasklistGetController {
 
       req.session.userCase = await client.retrieveByCaseId(caseId, citizenUser);
       mapDataInSession(req.session.userCase, citizenUser.id);
-
+      req.session.userCase.hearingCollection = [];
```
Comment (vivek-sekhar) [inline src/main/steps/tasklistresponse/TasklistGetController.ts:32]: why is this required?

Comment (SasankAchyuta9) [reply src/main/steps/tasklistresponse/TasklistGetController.ts:32]: Yes Removed.

File: src/main/steps/respondent/task-list/content.ts (hunk 26 above)
Comment (SasankAchyuta9) [reply src/main/steps/respondent/task-list/content.ts:1]: Yes Removed.

File: src/main/app/controller/GetController.ts (hunk 23 above)
Comment (SasankAchyuta9) [reply src/main/app/controller/GetController.ts:33]: Removed.

File: src/main/steps/common/yourhearings/hearings/content.ts [hunk 29]
```diff
@@ -59,8 +143,164 @@ export const form: FormContent = {
...
+          const endDate = hearing.hearingDaySchedule![len - 1].hearingStartDateTime!;
+          const endDateFormatted =
+            req.session.lang === 'cy'
+              ? dayjs(endDate).locale('cy').format('DD MMMM YYYY').toString()
+              : dayjs(endDate).locale('en').format('DD MMMM YYYY').toString();
+          dates = dates + ' - ' + endDateFormatted;
+        }
+        const lengthOfHearing = hearing.hearingDaySchedule?.length;
+        const hearingMethod = getHearingMethod(req, hearing.hearingDaySchedule![0].attendees!);
+        const hearingDays: object[] = [];
```
[the remark this answers is not in the document]
Comment (SasankAchyuta9) [reply src/main/steps/common/yourhearings/hearings/content.ts:197]: Yeah respective types are created in definition.ts

File: src/main/steps/common/yourhearings/hearings/content.ts [hunk 30]
```diff
@@ -59,8 +143,164 @@ export const form: FormContent = {
...
+            amPm,
+            durationInDayOrHours,
+            minutes,
+            judgeName,
+            venue,
+            address,
+            roomId,
+          });
+        }
+        req.session.userCase.futureHearings1.push({
```
[the remark this answers is not in the document]
Comment (SasankAchyuta9) [reply src/main/steps/common/yourhearings/hearings/content.ts:227]: yeah modified logic.

File: src/main/steps/common/yourhearings/hearings/content.ts (hunk 21 above)
Comment (SasankAchyuta9) [reply src/main/steps/common/yourhearings/hearings/content.ts:235]: yeah modified all the logic

File: src/main/steps/common/yourhearings/hearings/content.ts (hunk 22 above)
Comment (SasankAchyuta9) [reply src/main/steps/common/yourhearings/hearings/content.ts:153]: Yes modified all the logic

File: src/main/app/case/case.ts [hunk 31]
```diff
@@ -370,6 +370,13 @@ export interface Case {
   caseInvites?: CaseInvite[];
   orderCollection?: ListValue<PRLDocument>[];
   hearingCollection?: HearingsList[];
+  nextHearing?: HearingsList[];
+  nextHearing1?: object[];
```
[the remark this answers is not in the document]
Comment (SasankAchyuta9) [reply src/main/app/case/case.ts:374]: Yes removed.

File: src/main/steps/respondent/yourhearings/hearings/content.ts (hunk 27 above)
Comment (SasankAchyuta9) [reply src/main/steps/respondent/yourhearings/hearings/content.ts:16]: Yes modified the logic

File: src/main/steps/applicant/yourhearings/hearings/content.ts [hunk 32]
```diff
@@ -19,13 +19,13 @@ export const generateContent: TranslationFn = content => {
   } else {
     hearingsContent.linkforsupport = APPLICANT_ATTENDING_THE_COURT;
   }
-  request.session.userCase.hearingOrders = [];
+  const hearingOrders: object[] = [];
```
Comment (vivek-sekhar) [inline src/main/steps/applicant/yourhearings/hearings/content.ts:22]: put the right type for hearingOrders

File: src/main/steps/common/yourhearings/hearings/template.njk [hunk 33]
```diff
@@ -1,5 +1,6 @@
 {% extends "common/template.njk" %}
 {% extends "common/breadcrumb/template.njk" %}
+{% from "govuk/components/table/macro.njk" import govukTable %}
```
Comment (vivek-sekhar) [inline src/main/steps/common/yourhearings/hearings/template.njk:3]: Why govukTable is imported?

File: src/main/steps/respondent/yourhearings/hearings/content.ts [hunk 34]
```diff
@@ -7,13 +7,13 @@ export const generateContent: TranslationFn = content => {
   const hearingsContent = yourhearingshearingscontent(content);
   hearingsContent.linkforsupport = CA_DA_ATTENDING_THE_COURT;
   const request = content.additionalData?.req;
-  request.session.userCase.hearingOrders = [];
+  const hearingOrders: object[] = [];
```
Comment (vivek-sekhar) [inline src/main/steps/respondent/yourhearings/hearings/content.ts:10]: Put the right type for hearingOrders

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/common/yourhearings/hearings/template.njk (hunk 33 above)
Comment (SasankAchyuta9) [reply src/main/steps/common/yourhearings/hearings/template.njk:3]: Yeah sorry.. may be by mistake I have added this... I will remove this.

File: src/main/steps/respondent/yourhearings/hearings/content.ts (hunk 34 above)
Comment (SasankAchyuta9) [reply src/main/steps/respondent/yourhearings/hearings/content.ts]: Done this

File: src/main/steps/applicant/yourhearings/hearings/content.ts (hunk 32 above)
Comment (SasankAchyuta9) [reply src/main/steps/applicant/yourhearings/hearings/content.ts]: Done this.

**Review by vivek-sekhar — APPROVED** [review]

File: src/main/steps/applicant/yourhearings/hearings/content.ts (hunk 32 above)
Comment (SasankAchyuta9) [reply src/main/steps/applicant/yourhearings/hearings/content.ts]: Done this

**Review by AlokDatta — APPROVED** [review]
